src/util/utils.js

```diff
@@ -1,8 +1,86 @@
 /* eslint-disable max-classes-per-file */
+const http = require('http');
+const https = require('https');
+const { Resolver } = require('dns').promises;
+const fetch = require('node-fetch');
+
 const util = require('util');
 const logger = require('../logger');
 const stats = require('./stats');
 
+const resolver = new Resolver();
+// Cloudflare and Google dns
+resolver.setServers(['1.1.1.1', '8.8.8.8']);
+
+const LOCALHOST_IP = '127.0.0.1';
+const LOCALHOST_URL = `http://localhost`;
+const RECORD_TYPE_A = 4; // ipv4
+
+const staticLookup = (versionid) => async (hostname, _, cb) => {
+  let ips;
+  const resolveStartTime = new Date();
+  try {
+    ips = await resolver.resolve(hostname);
+  } catch (error) {
+    stats.timing('fetch_dns_resolve_time', resolveStartTime, { versionid, error: true });
+    cb(null, `unable to resolve IP address for ${hostname}`, RECORD_TYPE_A);
+    return;
+  }
+  stats.timing('fetch_dns_resolve_time', resolveStartTime, { versionid });
+
+  if (ips.length === 0) {
+    cb(null, `resolved empty list of IP address for ${hostname}`, RECORD_TYPE_A);
+    return;
+  }
+
+  for (const ip of ips) {
+    if (ip.includes(LOCALHOST_IP)) {
+      cb(null, `cannot use ${LOCALHOST_IP} as IP address`, RECORD_TYPE_A);
+      return;
+    }
+  }
+
+  cb(null, ips[0], RECORD_TYPE_A);
+};
+
+const staticDnsAgent = (scheme, versionId) => {
```

Comment:
Is below a better naming for this function ? `httpAgentWithDnsLookup()`
Wdyt?

Reply:
I would throw the exception here saying that invalid scheme provided, accepted only http or https !
Wdyt ?!

Reply:
renamed